langchain/schema/language_model.py

```diff
@@ -48,6 +69,161 @@ class BaseLanguageModel(Serializable, ABC):
     """
 
+    # --- Runnable methods ---
+
+    def _convert_input(self, input: LanguageModelInput) -> PromptValue:
+        if isinstance(input, PromptValue):
+            return input
+        elif isinstance(input, str):
+            return StringPromptValue(text=input)
+        elif isinstance(input, list):
+            return ChatPromptValue(messages=input)
+        else:
+            raise ValueError(
+                f"Invalid input type {type(input)}. "
+                "Must be a PromptValue, str, or list of BaseMessages."
+            )
+
+    def invoke(
+        self,
+        input: LanguageModelInput,
+        stop: Optional[List[str]] = None,
+        **kwargs: Unpack[RunnableConfig],
+    ) -> str:
+        return (
+            self.generate_prompt([self._convert_input(input)], stop=stop, **kwargs)
+            .generations[0][0]
+            .text
+        )
+
+    async def ainvoke(
+        self,
+        input: LanguageModelInput,
+        stop: Optional[List[str]] = None,
+        **kwargs: Unpack[RunnableConfig],
+    ) -> str:
+        llm_result = await self.agenerate_prompt(
+            [self._convert_input(input)], stop=stop, **kwargs
+        )
+        return llm_result.generations[0][0].text
+
+    def batch(
+        self,
+        inputs: List[LanguageModelInput],
+        config: Optional[RunnableConfig | List[RunnableConfig]] = None,
+        max_concurrency: Optional[int] = None,
+    ) -> List[str]:
+        if isinstance(config, list):
+            config = config[0]
+        if config is None:
+            config = {}
+
+        llm_result = self.generate_prompt(
+            [self._convert_input(input) for input in inputs], **config
+        )
+        return [g[0].text for g in llm_result.generations]
+
+    async def abatch(
+        self,
+        inputs: List[LanguageModelInput],
+        config: Optional[RunnableConfig | List[RunnableConfig]] = None,
+        max_concurrency: Optional[int] = None,
+    ) -> List[str]:
+        if isinstance(config, list):
+            config = config[0]
+        if config is None:
+            config = {}
+
+        llm_result = await self.agenerate_prompt(
+            [self._convert_input(input) for input in inputs], **config
+        )
+        return [g[0].text for g in llm_result.generations]
+
+    def stream(
+        self,
+        input: LanguageModelInput,
+        stop: Optional[List[str]] = None,
+        **kwargs: Unpack[RunnableConfig],
+    ) -> Iterator[str]:
+        if not hasattr(self, "streaming"):
+            # model doesn't support streaming, so use default implementation
+            yield self.invoke(input, stop=stop, **kwargs)
+        else:
+            from langchain.callbacks.streaming_iter import IteratorCallbackHandler
+
+            # enable streaming, if it's not already enabled
+            original_streaming = cast(bool, self.streaming)  # type: ignore
+            self.streaming = True
```

Comment:
mmmmmmmm

Reply:
how many models support streaming atm? would it be very hard/annoying to implement an explicit _stream function on them and just call that here (and default implementation can be `yield self.invoke(input, stop=stop, **kwargs)`